sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessor.java

```diff
@@ -237,6 +237,7 @@ public BatchSpansProcessor build() {
 
   private static Thread newThread(Runnable runnable) {
     Thread thread = MoreExecutors.platformThreadFactory().newThread(runnable);
+    thread.setDaemon(true);
```

Comment:
I did not put this inside the try/catch(SecurityException) on purpose because I think failing to daemonize the thread should be a fatal error.

Reply:
I don't know if I agree with that. Not having the threads be daemons is only important in a narrow use-case. If the lifecycles are being managed correctly, then `shutdown` will get called appropriately, and everything will be fine.

Not putting this in the try/finally means that this will not be usable at all in the secure environment, even if they are managing the lifecycle as designed.  I think this should be inside the try block.

Reply:
Why is it insecure to set a thread name or make a thread a daemon? Also, we do not have documentation anywhere that some "shutdown" should be called (there is also no required "initialize" call that would make this more obvious). Can we get around this with a custom thread factory that sets the daemon flag already in the thread constructor?

If we can test with one of this API-restricted environments, that would make sense.

Reply:
No idea, but the javadoc says that a security manager can forbid it.
```
     * @throws  SecurityException
     *          if {@link #checkAccess} determines that the current
     *          thread cannot modify this thread
     */
    public final void setDaemon(boolean on) {
```

Reply:
I'm also all-for adding documentation about the strong-suggestion that `shutdown` should be called on your TracerSdkProvider/span processors.

Reply:
I think the `SecurityException` is only checked/thrown if you try to modify a JVM system thread.

Reply:
unless someone has some super strict custom security manager, maybe?

Reply:
Oh, I found actual doc, check out https://docs.oracle.com/javase/8/docs/api/java/lang/SecurityManager.html#checkAccess-java.lang.Thread-

Reply:
right, so out of the box, it's a no-op, but if someone has a strict custom security manager, it could fail. how do we want this to behave in that case?

Reply:
> If the thread argument is not a system thread, this method just returns silently

Reply:
oh nevermind 🤦

Reply:
(I had stopped reading the javadoc once I saw what I wanted to see 😆)

Reply:
My opinion: we shouldn't fail if they have a strict security manager, so we need to move the `setDaemon` call inside the try.

Reply:
I'm trying to think what I've seen other libraries do.

Typically you give them an `ExecutorService`, so you configure it how you want.

Maybe accept `ThreadFactory` as a builder option?

And use `Executors.defaultThreadFactory()` as the default (non-daemon).

Reply:
In OpenCensus one example where name and daemon could fail was appengine.

Reply:
Note that MoreExecutors uses a special thread factory in AppEngine according to it's docs. That may be the reason ... or might mean that this does not happen with MoreExecutors 🤷

Reply:
Either way, the safest thing is to put the `setDaemon` call inside the `try`. Thanks!

Reply:
Or don't set it to daemon, and allow user to pass in `ThreadFactory`.

Reply:
also a fine solution; a bit of a bigger change than this PR, but I'd be fine with it, as well.

Reply:
I'd be up for going with @jkwatson's solution for now - we can follow up @trask approach in another PR ;)

Reply:
I would suggest for the moment to put this in the try-catch as we have the name.